sdks/python/apache_beam/runners/portability/fn_api_runner/translations.py

```diff
@@ -357,12 +357,29 @@ def wrapper(self, *args):
 
 class TransformContext(object):
 
-  _KNOWN_CODER_URNS = set(
+  _COMMON_CODER_URNS = set(
       value.urn for (key, value) in common_urns.coders.__dict__.items()
       if not key.startswith('_')
       # Length prefix Rows rather than re-coding them.
   ) - set([common_urns.coders.ROW.urn])
 
+  _REQUIRED_CODER_URNS = set([
+      common_urns.coders.WINDOWED_VALUE.urn,
+      # For impulse.
+      common_urns.coders.BYTES.urn,
+      common_urns.coders.GLOBAL_WINDOW.urn,
+      # For GBK.
+      common_urns.coders.KV.urn,
+      common_urns.coders.ITERABLE.urn,
+      # For SDF.
+      common_urns.coders.DOUBLE.urn,
```

Comment:
This feels a little weird. Should we bring requirements of specific transforms (even though primitive) into generic requirements? I'd say, that the required coders should be the set without which *no* pipeline will ever successfully run. A pipeline using SDF might need DoubleCoder, but in such case both PIpeline (SDK) and the runner will have it in its supported components, right?

Reply:
One could, I suppose, do away with the notion of required coders, and rely on the fact that SDKs will provide them (so the intersection would contain whatever is necessary). I do think it simplifies things to be able to just assume that, say, KV coder is around rather than wondering if/when one needs to length prefix it in the implementation of various primitives. 

This also guards against poor behavior in case some SKD doesn't populate this field...

Reply:
Alright, at the end, this is essentially the same case as with KV and Iterable for GBK.